carpooling/models/models.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+
+class VehicleTrip(models.Model):
+    _name = "carpooling.vehicle_trip"
+    _description = "carpooling.vehicle_trip.description"
+
+    driver = fields.Many2one("res.users", "Driver", required=True, default=lambda self: self.env.user)
+    driver_uid = fields.Integer(compute="_get_driver_uid", store=True)
+
+    # field_lat = fields.Float(string="Latitude")
+    # field_lng = fields.Float(string="Longitude")
+
+    @api.depends('driver')
+    def _get_driver_uid(self):
+        for record in self:
+            record.driver_uid = record.driver.id
+
+    is_current_user = fields.Boolean(compute="_is_current_user_driver")
+
+    @api.depends('driver')
+    def _is_current_user_driver(self):
+        for record in self:
+            record.is_current_user = (record.driver == self.env.user)
+
+    vehicle_type = fields.Selection(
+        string='Vehicle type',
+        selection=[('small_car', 'Small car'), ('medium_car', 'Medium car'), ('big_car', 'Big car'), ('suv', 'SUV'), ('minivan', 'Minivan'), ('minivan', 'Minivan')],
+        help="Different possible types of vehicle"
+    )
+
+    departure_loc = fields.Char(string="Departure location", required=True)
+    destination_loc = fields.Char(string="Destination location", required=True)
+    departure_time = fields.Datetime(string="Departure time", required=True)
+    expired = fields.Boolean(compute="_compute_expired", store=True)
+    @api.depends("departure_time")
+    def _compute_expired(self):
+        for record in self:
+            now = datetime.now()
+            record.expired = record.departure_time < now
+                
+
+    available_seats = fields.Integer(required=True, string="Available seats")
+    remaining_seats = fields.Char(compute='_remaining_seats', string="Remaining seats")
+    remaining_seats_int = fields.Integer(compute='_remaining_seats_int', string="Remaining seats")
+    
+    description = fields.Text()
+
+    passengers = fields.Many2many(comodel_name="res.users", string="Passengers")
+
+    @api.constrains('passengers')
+    def _check_number_of_passengers(self):
```

Comment:
Clean, but as said by Romain, you can **also** add a domain on the view , it's more user friendly .
But the constraint is mandatory as you want to add a layer of security here